### PR TITLE
LINK-2138 | GDPR removal improvements

### PR DESCRIPTION
**Pull request by nicobav**
Adds check logic for deletion of on going payments and upcoming signups

Refs LINK-2138

### CODE REVIEW

File: helevents/utils.py [hunk 1]
```diff
@@ -45,6 +48,54 @@ def delete_user_and_gdpr_data(
             ]
         )
 
+    minimum_event_end = timezone.now() - timezone.timedelta(
+        days=settings.GDPR_API_DELETE_EVENT_END_TRESHOLD_DAYS
+    )
+    has_upcoming_signups = user.signup_created_by.filter(
+        signup_group__registration__event__end_time__gt=minimum_event_end
+    ).exists()
```
[the remark this answers is not in the document]
Comment (nicobav) [reply helevents/utils.py:56]: very nicely catched 🎣 I'll add the check

File: helevents/utils.py [hunk 2]
```diff
@@ -45,6 +49,55 @@ def delete_user_and_gdpr_data(
             ]
         )
 
+    minimum_event_end = timezone.now() - timezone.timedelta(
+        days=settings.GDPR_API_DELETE_EVENT_END_TRESHOLD_DAYS
```
Comment (tuomas777) [inline helevents/utils.py:53]: Don't worry, 3/4 is still a pretty good result! 😄  

```suggestion
        days=settings.GDPR_API_DELETE_EVENT_END_THRESHOLD_DAYS
```

Comment (nicobav) [reply helevents/utils.py]: Yes, I consider that an almost victory 🥈

**Review by tuomas777 — APPROVED** [review]
LGTM 👍 🐴

**Pull request merged**